File: _tbx/supportfct/plotirf1.m
function plotirf1(irf, irfbands, printvars, path)

H = size(irf,1)-1;
n = size(irf,2);
lo = irfbands(:,:,1);
up = irfbands(:,:,2);
nrow = ceil(n/3);


figure()
for vv = 1:n
    maxim = max([irf(:,vv); lo(:,vv); up(:,vv)]);
    maxim = maxim + 0.1*abs(maxim);
    minim = min([irf(:,vv); lo(:,vv); up(:,vv)]);
    minim = minim - 0.1*abs(minim);
    subplot(nrow, 3, vv)
    plot(0:H, irf(:,vv), 'k', 'LineWidth', 2)
    hold on
    plot(0:H, zeros(H+1,1), 'k', 'LineWidth', 0.5)
    plot(0:H, lo(:,vv), '--k')
    plot(0:H, up(:,vv), '--k')
    grid on
    axis('tight')
    ylim([minim maxim])
    title(printvars(vv),'FontSize',14)
    set(gca,'GridLineStyle','-','Layer','bottom')
    if H <= 30
        set(gca,'XTickLabelMode', 'manual','XTickLabel',{'0','4','8','12','16','20','24'},'XTick',[0:4:24])
    else
        set(gca,'XTickLabelMode', 'manual','XTickLabel',{'0','12','24','36','48'},'XTick',[0:12:48])
    end
    set(gcf,'paperpositionmode','auto')
end
set(gcf, 'position', [0 0 800 nrow*200]);
print(gcf,'-dpng','-loose',strcat(path,'_irf.png'));

File: _tbx/supportfct/plotirf1.R
plotirf1 <- function(irf, irfbands, printvars){
  
  H <- nrow(irf)-1
  n <- ncol(irf)
  lo <- irfbands[,,1]
  up <- irfbands[,,2]
  numrows <- ceiling(n^(1/2))

  par(mfrow=c(numrows,ceiling(n/numrows)), mai = c(0.4, 0.4, 0.4, 0.4))
  for (vv in 1:n){
    maxim <- max(c(irf[,vv]), lo[,vv], up[,vv])
    maxim <- maxim + 0.1*abs(maxim)
    minim <- min(c(irf[,vv]), lo[,vv], up[,vv])
    minim <- minim - 0.1*abs(minim)
    plot(0:H, irf[,vv], 'l', lwd=2, xaxt='n', xaxs='i',
         xlab='', ylab='', main=printvars[vv], ylim=c(minim, maxim))
    grid()
    abline(h=0, lwd=0.5)
    lines(0:H, lo[,vv], lty=2)
    lines(0:H, up[,vv], lty=2)
    if (H <= 30){
      axis(1, at=seq(0, H, 4), labels=seq(0, H, 4))
    } else {
      axis(1, at=seq(0, H, 12), labels=seq(0, H, 12)) 
    }
    box()
  }
  par(mfrow=c(1,1), mar=c(4.6, 4.1, 2.1, 2.1))
}


